LIReC/create_db.py

```diff
@@ -0,0 +1,80 @@
+import re
+from decimal import Decimal
+from config import get_connection_string
+import os
+from LIReC.lib import models, db_access
+from LIReC.lib.calculator import Constants
+
+COMMAND = 'psql {connection_string} < db/lib/db/create_db.sql'
+MIN_PRECISION = 20
+
+def get_manual_value(path): # works with the OEIS format, credit where credit is due
+    lines = open(path)
+    value = ''
+    first_index = None
+    precision = None
+    while True:
+        line = lines.readline()
+        if not line:
+            return value, int(precision)
+        res = re.match('\\s*(\\d+)\\s*(\\d+)\\s*', line)
+        if not res:
+            continue
+        if not value:
+            first_index = int(res.group(1)) # this is the number of digits before the decimal point!
+        if first_index == 0:
+            if not value:
+                value = '0'
+            value += '.'
+        value += res.group(2)
+        first_index -= 1
+        precision = res.group(1)
+
+if __name__ == '__main__':
+    os.system(COMMAND.format(connection_string=get_connection_string(db_name='postgres')))
+    
+    precision = 4000
+    print(f'Using {precision} digits of precision')
+    Constants.set_precision(precision)
+    db = db_access.LIReC_DB()
+    for x in Constants.__dict__.keys():
+        if x[0] == '_' or x == 'set_precision':
+            continue
+        print(f'Adding named constant {x}')
+        named_const = models.NamedConstant()
+        const_func = eval(f'Constants.{x}')
```

Comment:
Good call, no idea why I didn't think of that since I already call `__dict__`. There's another part of the code that also uses `eval`, which I can modify to take from `__dict__` instead in a similar way, so I'll do both.

Reply:
Using __dict__ is also probably not the best afaik, but it would require changing how the Constants file is laid out...

Reply:
Is there a better way? I don't think converting `calculator.py` to a massive switch case or a massive dictionary would be better... There's also the other code I mentioned, and getting rid of calls to `eval` or `__dict__` there would probably involve just circumventing them with a literal dictionary of string to class, which kinda sounds dumb...